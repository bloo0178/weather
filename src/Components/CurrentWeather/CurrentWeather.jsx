import React from 'react';
import getCurrent from '../../api/currentWeatherAPI';
import { Card, CardBody, CardTitle } from 'reactstrap';
import styles from './CurrentWeather.module.scss';

class CurrentWeather extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            date: '',
            temp: '',
            icon: '',
            description: '',
        }
    }

    componentDidUpdate = (prevProps) => {
        if (this.props.location !== prevProps.location) {
            getCurrent(this.props.location, this.props.units, (currentWeather) => {
                this.setState({
                    date: currentWeather.date,
                    temp: currentWeather.temp,
                    icon: currentWeather.icon,
                    description: currentWeather.description
                })
            });
        }
    }

    componentDidMount() {
        getCurrent(this.props.location, this.props.units,
            (currentWeather) => {
                this.setState({
                    date: currentWeather.date,
                    temp: currentWeather.temp,
                    icon: currentWeather.icon,
                    description: currentWeather.description
                })

            })
    }

    render() {
        return (
            <div className={styles.cardWrapper}>
                <Card className="border-0">
                    <CardBody className="text-center">
                        <CardTitle>Currently</CardTitle>
                        {/*<div>{this.state.date}</div>*/}
                        <img src={this.state.icon} alt="Current weather icon" />
                        <div>{this.state.temp + ' ' + this.props.tempUnit}</div>
                        <div>{this.state.description}</div>
                    </CardBody>
                </Card>
            </div>
        )
    }
}


export default CurrentWeather; 