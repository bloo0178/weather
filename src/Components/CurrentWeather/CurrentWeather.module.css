.Container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-bottom: 1rem;
}
