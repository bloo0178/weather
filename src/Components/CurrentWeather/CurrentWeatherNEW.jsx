import React from "react";
import getCurrent from "../../api/currentWeatherAPI";
//import { Card, CardBody, CardTitle } from "reactstrap";
import { TimeCard } from "../TimeCard/TimeCard";
import styles from "./CurrentWeatherNEW.module.scss";

class CurrentWeatherNEW extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      date: "",
      temp: "",
      icon: "",
      description: ""
    };
  }

  componentDidUpdate(prevProps) {
    const { location, units } = this.props;
    if (location !== prevProps.location) {
      getCurrent(location, units, currentWeather => {
        this.setState({
          date: currentWeather.date,
          temp: currentWeather.temp,
          icon: currentWeather.icon,
          description: currentWeather.description
        });
      });
    }
  }

  componentDidMount() {
    const { location, units } = this.props;
    getCurrent(location, units, currentWeather => {
      this.setState({
        date: currentWeather.date,
        temp: currentWeather.temp,
        icon: currentWeather.icon,
        description: currentWeather.description
      });
    });
  }

  render() {
    const { icon, temp, description } = this.state;
    const { tempUnit } = this.props;
    if (!icon) {
      return <div>Loading...</div>;
    }
    const iconSrc = require(`../../common/icons/${icon}.svg`);
    return <TimeCard temp={temp} tempUnit={tempUnit} iconSrc={iconSrc} time="Currently" />;
  }
}

export default CurrentWeatherNEW;
