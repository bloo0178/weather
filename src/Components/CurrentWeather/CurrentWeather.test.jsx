import React from "react";
import CurrentWeather from "./CurrentWeather";
import { shallow } from "enzyme";

describe("<CurrentWeather />", () => {
	test("renders correctly", () => {
		let wrapper = shallow(
			<CurrentWeather currentWeather={{ icon: "01d", temp: "85" }} />
		);
		console.log(wrapper.debug());
		expect(wrapper).toMatchSnapshot();
	});
});
