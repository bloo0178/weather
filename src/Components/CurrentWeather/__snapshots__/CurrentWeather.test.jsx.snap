// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentWeather /> renders correctly 1`] = `
<div
  className="Container"
>
  <TimeCard
    iconSrc={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "01d.svg",
      }
    }
    temp="85"
    time="Currently"
  />
</div>
`;
